ci-operator/step-registry/aws/provision/tags-for-byo-vpc/aws-provision-tags-for-byo-vpc-commands.sh

```diff
@@ -13,24 +13,32 @@ fi
 
 REGION="${LEASED_RESOURCE}"
 
-infra_id=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
 vpc_id=$(head -n 1 ${SHARED_DIR}/vpc_id)
 private_subnet_ids=$(yq-go r -j ${SHARED_DIR}/private_subnet_ids | jq -r '[ . | join(" ") ] | @csv' | sed "s/\"//g")
 public_subnet_ids=$(yq-go r -j ${SHARED_DIR}/public_subnet_ids | jq -r '[ . | join(" ") ] | @csv' | sed "s/\"//g")
-
-if [[ -z $vpc_id ]] || [[ -z $private_subnet_ids ]] || [[ -z $infra_id ]]; then
+if [[ -z $vpc_id ]] || [[ -z $private_subnet_ids ]]; then
   echo "Error: Can not get VPC id or private subnets, exit"
   echo "vpc: $vpc_id, private_subnet_ids: $private_subnet_ids"
   exit 1
 fi
 
-echo "Adding tags for VPC: $vpc_id, tags: kubernetes.io/cluster/${infra_id}, value: shared."
-aws --region $REGION ec2 create-tags --resources $vpc_id --tags Key=kubernetes.io/cluster/${infra_id},Value=shared
+# Add tags on VPC
+if [[ ${MANAGED_CLUSTER} == "no" ]]; then
+  infra_id=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
+  if [[ -z $infra_id ]]; then
+    echo "Error: infra_id is empty"
+    exit 1
+  fi
+
+  echo "Adding tags for VPC: $vpc_id, tags: kubernetes.io/cluster/${infra_id}, value: shared."
+  aws --region $REGION ec2 create-tags --resources $vpc_id --tags Key=kubernetes.io/cluster/${infra_id},Value=shared
+fi
 
+# Add tags on subnets
```

Comment:
```suggestion
# Add tags on private subnets
```

Reply:
Below there is a step to add the public subnets, so `Add tags on subnets` should be more reasonable.